python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -331,17 +339,54 @@ def _get_records(self, graphene_info: ResolveInfo) -> Sequence[RunRecord]:
 
     def _get_partition_run_data(self, graphene_info: ResolveInfo) -> Sequence[RunPartitionData]:
         if self._partition_run_data is None:
-            self._partition_run_data = (
-                graphene_info.context.instance.run_storage.get_run_partition_data(
-                    runs_filter=RunsFilter(
-                        tags={
-                            BACKFILL_ID_TAG: self._backfill_job.backfill_id,
-                        }
+            # Storage layer not equipped to calculate partition status for ranged job backfills so
+            # we perform the calculation ad hoc here.
+            partition_set = self._get_partition_set(graphene_info)
```

Comment:
Can we do early returns to avoid excessive nesting?

Going to req changes because I want to use this as an example for a code smell.